### ISSUE TITLE
"Go to Symbol in Workspace…" is slow with moderately sized solution containing only C# projects

### ISSUE THREAD

**Issue by dmgonch** [OmniSharp/omnisharp-roslyn#1243]
REPRO: I opened a solution in VS Code containing 68 C# projects (according to the response for "/projects" request in OmniSharp output log). Once all projects were loaded (took about a minute), I pressed Ctrl-T and had to wait for at least 15 seconds for the list to be populated. This happened every time Ctrl-T is pressed - not only the first time. Additionally, typing a single character freezes the list again for few seconds which makes the feature only marginally useful in this case.

EXPECTED: IMO OmniSharp could:
 - Cache results of recently issued "/findsymbols" requests.
 - Apply a strict timeout when processing "/findsymbols" requests and return cached or partial or simply empty result when the timeout is reached.

**Comment by SirIntruder** [comment]
IIRC Problem is in vscode rendering thousands of symbols in the ui after server returns them, not on the server side.

**Comment by dmgonch** [comment]
Interesting. I was only judging by OmniSharp output log because I see a significant delay after "/findsymbols" request printed and before the response appears. In any case I would argue that "/findsymbols" processing should be limited in both time and the amount of data it is allowed to return. A user IMO is unlikely to just scroll through thousands of entries and s/he is more likely to continue typing the filter to narrow the result to only few entries.

**Comment by dmgonch** [comment]
BTW, this issue I opened sounds like a dupe of https://github.com/OmniSharp/omnisharp-vscode/issues/1808

**Comment by SirIntruder** [comment]
Yup that's it. I remember I just modified "workspaceSymbolProvider.js" manually so I never pass on insanely big symbol lists to vscode, and it solved all the problems. Wanted to make a PR but it just flew over my head.

**Comment by dmgonch** [comment]
This is definitely could be a good quick fix, but long term IMO ideally each supported file system (MSBuild, Cake, etc) limits as well how many items it attempts to discover and send back with /findsymbols response.
